stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,292 @@
+                pick_type.default_location_dest_id = pack_loc
+                pack_type.default_location_src_id = pack_loc
+                pack_type.default_location_dest_id = output_loc
+        # Enable relevant locations (pack/output) based on the delivery steps
+        self.warehouse_id._update_location_delivery(self.delivery_steps)
+        try:
+            with self.env.cr.savepoint():
+                wh_vals = {
+                    "delivery_steps": self.delivery_steps,
+                    "out_type_id": out_type.id,
+                }
+                if pick_type:
+                    wh_vals.update(
+                        {
+                            "pick_type_id": pick_type.id,
+                            "wh_output_stock_loc_id": output_loc.id,
+                        }
+                    )
+                if pack_type:
+                    wh_vals.update(
+                        {
+                            "pack_type_id": pack_type.id,
+                            "wh_pack_stock_loc_id": pack_loc.id,
+                        }
+                    )
+                wh_vals["out_type_id"] = out_type.id
+                self.warehouse_id.write(wh_vals)
+                vals = self.warehouse_id.delivery_route_id.copy_data()[0]
+                raise Exception()
+        except Exception:
+            pass
+        vals.update(
+            {
+                "name": f"{self.warehouse_id.name}: {self.name}",
+                "active": True,
+                "flow_id": self.id,
+                # Give the priority to routes/rules not tied to a flow
+                # when the initial move is created by the procurement
+                "sequence": 1000,
+            }
+        )
+        self.delivery_route_id = self.env["stock.location.route"].create(vals)
+        self.to_picking_type_id = out_type
+        # Subscribe the route to the warehouse
+        if self.delivery_route_id.warehouse_selectable:
+            self.warehouse_id.route_ids |= self.delivery_route_id
+
+    def action_generate_route(self):
+        for flow in self:
+            flow._generate_delivery_route()
+        return True
+
+    def _search_for_move_domain(self, move):
+        return [
+            ("from_picking_type_id", "=", move.picking_type_id.id),
+            ("carrier_ids", "in", move.group_id.carrier_id.ids),
+        ]
+
+    @api.model
+    def _search_for_move(self, move):
+        """Return the first flow matching the move."""
+        domain = self._search_for_move_domain(move)
+        flow = self.search(domain, limit=1)
+        if flow and move.filtered_domain(safe_eval(flow.move_domain or "[]")):
+            return flow
+        return self.browse()
+
+    def _apply_on_move(self, move):
+        """Apply the flow configuration on the move."""
+        if not self:
+            return False
+        logger.info("Applying flow '%s' on '%s'", self.name, move)
+        rule = self.delivery_route_id.rule_ids.filtered(
```

Comment:
If no rule is found, an error must be raised
A warning on the flow form would be also helpful in this case